packagehandlers/packagehandlers_test.go

```diff
@@ -498,7 +498,13 @@ func TestGetProjectPoms(t *testing.T) {
 	defer func() {
 		assert.NoError(t, os.Chdir(currDir))
 	}()
-	assert.NoError(t, mvnHandler.getProjectPoms())
+	depTreeOutput, clearMavenDepTreeRun, err := mvnHandler.RunMavenDepTree()
+	assert.NoError(t, err)
+	err = mvnHandler.getProjectPoms(depTreeOutput)
+	assert.NoError(t, err)
+	if assert.NotNil(t, clearMavenDepTreeRun) {
+		defer assert.NoError(t, clearMavenDepTreeRun())
```

Comment:
defer only defers the outer func which is assert.NoError()

all the inners funcs will be triggered when the code will pass this line